brotli-sys/build.rs

```diff
@@ -16,33 +16,29 @@ fn main() {
 
     gcc::Config::new()
         .include("brotli/dec")
+        .file("brotli/common/dictionary.c")
         .file("brotli/dec/bit_reader.c")
         .file("brotli/dec/decode.c")
-        .file("brotli/dec/dictionary.c")
         .file("brotli/dec/huffman.c")
         .file("brotli/dec/state.c")
         .compile("libbrotli-dec.a");
 
-    // Omit "brotli/enc/dictionary.cc" from the source list in order to avoid
-    // multiple definition linker errors with "brotli/dec/dictionary.c".
     gcc::Config::new()
-        .cpp(true)
         .include("brotli/enc")
-        .include("src")
-        .file("src/brotli_capi.cc")
-        .file("brotli/enc/backward_references.cc")
-        .file("brotli/enc/block_splitter.cc")
-        .file("brotli/enc/brotli_bit_stream.cc")
-        .file("brotli/enc/compress_fragment.cc")
-        .file("brotli/enc/compress_fragment_two_pass.cc")
-        .file("brotli/enc/encode.cc")
-        .file("brotli/enc/encode_parallel.cc")
-        .file("brotli/enc/entropy_encode.cc")
-        .file("brotli/enc/histogram.cc")
-        .file("brotli/enc/literal_cost.cc")
-        .file("brotli/enc/metablock.cc")
-        .file("brotli/enc/static_dict.cc")
-        .file("brotli/enc/streams.cc")
-        .file("brotli/enc/utf8_util.cc")
+        .file("brotli/enc/backward_references.c")
```

Comment:
Yup - looks like they can.

We do get multiple definition errors for `BROTLI_ALLOC` and `BROTLI_FREE` if we `#include` both [dec/port.h](https://github.com/google/brotli/blob/master/dec/port.h#L152-L157) and [enc/memory.h](https://github.com/google/brotli/blob/master/enc/memory.h#L42-L48) which will probably require skipping more checks in the systest unfortunately.